dicom_util/src/main/java/com/google/cloud/healthcare/DicomWebClientJetty.java

```diff
@@ -0,0 +1,222 @@
+package com.google.cloud.healthcare;
+
+import com.github.danieln.multipart.MultipartInput;
+import com.google.auth.oauth2.OAuth2Credentials;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.dcm4che3.net.Status;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.http.HttpVersion;
+import org.eclipse.jetty.http.MetaData;
+import org.eclipse.jetty.http.MetaData.Response;
+import org.eclipse.jetty.http2.api.Session;
+import org.eclipse.jetty.http2.api.Stream;
+import org.eclipse.jetty.http2.api.server.ServerSessionListener;
+import org.eclipse.jetty.http2.client.HTTP2Client;
+import org.eclipse.jetty.http2.frames.DataFrame;
+import org.eclipse.jetty.http2.frames.HeadersFrame;
+import org.eclipse.jetty.http2.frames.ResetFrame;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.FuturePromise;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class DicomWebClientJetty implements IDicomWebClient {
+
+  private static final int CONNECT_PORT = 443;
+
+  private final String serviceUrlPrefix;
+  private final OAuth2Credentials credentials;
+
+  public DicomWebClientJetty(
+      OAuth2Credentials credentials,
+      String serviceUrlPrefix) {
+    this.credentials = credentials;
+    this.serviceUrlPrefix = Util.trim(serviceUrlPrefix);
+  }
+
+  @Override
+  public MultipartInput wadoRs(String path) throws DicomWebException {
+    throw new UnsupportedOperationException("Not Implemented, use DicomWebClient");
+  }
+
+  @Override
+  public JSONArray qidoRs(String path) throws DicomWebException {
+    throw new UnsupportedOperationException("Not Implemented, use DicomWebClient");
+  }
+
+  @Override
+  public void stowRs(String path, InputStream in) throws DicomWebException {
+    try {
+      HTTP2Client client = new HTTP2Client();
+      SslContextFactory sslContextFactory = new SslContextFactory.Client();
+      client.addBean(sslContextFactory);
+      client.start();
+
+      HttpURI uri = new HttpURI(serviceUrlPrefix + "/" + Util.trim(path));
+
+      FuturePromise<Session> sessionPromise = new FuturePromise<>();
+      client.connect(sslContextFactory, new InetSocketAddress(uri.getHost(), CONNECT_PORT),
+          new ServerSessionListener.Adapter(), sessionPromise);
+      Session session = sessionPromise.get(5, TimeUnit.SECONDS);
```

Comment:
5 seconds might be too low for very large files. can we increase this limit to like 10 minutes?

Reply:
It's timeout to establish the session, it doesn't affect data transmission.